packages/flame/lib/src/geometry/shapes/line_segment.dart

```diff
@@ -0,0 +1,78 @@
+import 'dart:math';
+import 'dart:ui';
+
+import 'package:meta/meta.dart';
+import 'package:vector_math/vector_math_64.dart';
+
+import '../../game/transform2d.dart';
+import 'shape.dart';
+
+/// A line segment between points [from] and [to]. The points must be distinct.
+///
+/// The segment is undirected, meaning a `LineSegment(from, to)` is considered
+/// to be equivalent to a `LineSegment(to, from)`.
+@immutable
+class LineSegment extends Shape {
+  LineSegment(this.from, this.to)
+      : assert(from != to, 'Two distinct points required for a LineSegment');
+
+  final Vector2 from;
+  final Vector2 to;
+
+  @override
+  bool get isClosed => false;
+
+  @override
+  bool get isConvex => true;
+
+  @override
+  Vector2 get center => (from + to)..scale(0.5);
+
+  @override
+  double get perimeter => (to - from).length;
+
+  @override
+  Aabb2 get aabb {
+    final minX = min(from.x, to.x);
+    final maxX = max(from.x, to.x);
+    final minY = min(from.y, to.y);
+    final maxY = max(from.y, to.y);
+    return Aabb2.minMax(Vector2(minX, minY), Vector2(maxX, maxY));
```

Comment:
It should be possible to create this `Aabb2` without creating two new vectors I think?

Reply:
I don't see how, `Aabb2` has only the following constructors:
- `Aabb2.zero()`
- `Aabb2.copy(Aabb2 other)`
- `Aabb2.minMax(Vector2 min, Vector2 max)`
- `Aabb2.centerAndHalfExtents(Vector2 center, Vector2 halfExtents)`
- `Aabb2.fromBuffer(ByteBuffer buffer, int offset)`

Reply:
Alright starbge decision of them to not include a cobstructor from primitives, this wont be a concern when I've created the vector2 pool anyways.